Documentation: update consensus params in documentation specs #8502 (#8955)
tendermint/tendermint: spec/core/data_structures.md

## ConsensusParams

| Name      | Type                                | Description                                                                               | Field Number |
|-----------|-------------------------------------|-------------------------------------------------------------------------------------------|--------------|
| block     | [BlockParams](#blockparams)         | Parameters limiting the size of a block and time between consecutive blocks.              | 1            |

> | evidence  | [EvidenceParams](#evidenceparams)   | Parameters limiting the validity of evidence of byzantine behaviour.                      | 2            |

Reviewer: ```suggestion
| evidence  | [EvidenceParams](#evidenceparams)   | Parameters limiting the validity of evidence of Byzantine behaviour.                      | 2            |
```

| validator | [ValidatorParams](#validatorparams) | Parameters limiting the types of public keys validators can use.                          | 3            |

> | version   | [BlockParams](#blockparams)         | The ABCI application version.                                                             | 4            |

Reviewer 2: It looks like we don't have a subheader for the `VersionParams`

> | version   | [BlockParams](#blockparams)         | The ABCI application version.                                                             | 4            |

Author: https://github.com/tendermint/tendermint/pull/8955/files#diff-25f6a26c2e1cd554551a691c7caa4c3e3760b0bd5e29da1da858e71deadefb08L448-L452

Is this not the subheader you are referring to?

> | synchrony | [SynchronyParams](#synchronyparams) | Are used to define the validity of block timestamps.                               | 5            |

Reviewer: More details here: https://github.com/tendermint/tendermint/tree/master/spec/consensus/proposer-based-timestamp.

(This a suggestion, add a link to the PBTS spec)

> | synchrony | [SynchronyParams](#synchronyparams) | SynchronyParams influence the validity of block timestamps.                               | 5            |

Reviewer: ```suggestion
| synchrony | [SynchronyParams](#synchronyparams) | Parameters to define the validity of block timestamps.                               | 5            |
```

Another suggestion.

> | timeout   | [TimeoutParams](#timeoutparams)     | TimeoutParams configure the timings of the steps of the Tendermint consensus algorithm.   | 6            |

Reviewer: ```suggestion
| timeout   | [TimeoutParams](#timeoutparams)     | Configure the timings of the steps of the Tendermint consensus algorithm.   | 6            |
```

> | ABCI      | [ABCIParams](#abciparams)           | ABCIParams configure ABCI functionality specific to the Application Blockchain Interface. | 7            |

Reviewer: This description is a little redundant, isn't it? ABCI = Application BlockChain Interface.

Reviewer 2: https://www.youtube.com/shorts/We4-HDdkPmQ

### HashedParams

> Hashed Params is a subset of Consensus Params, it is amino encoded and hashed into the Header.ConsensusHash.

Reviewer 2: Is it actually amino-encoded?

Reviewer 2: https://github.com/tendermint/tendermint/blob/65c0fba564a023fe097ec70a1f652664957e6ce3/types/params.go#L385 

Just checked - it's proto encoded.

| Name            | Type  | Description                                                                                                                                                                                                 | Field Number |
|-----------------|-------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------|

> | block_max_bytes | int64 | Max size of a block, in bytes.                                                                                                                                                                              | 1            |

Reviewer 3: ```suggestion
| block_max_bytes | int64 | Maximum size of a block, in bytes.                                                                                                                                                                          | 1            |
```

> | block_max_gas   | int64 | Max sum of `GasWanted` in a proposed block. NOTE: blocks that violate this may be committed if there are Byzantine proposers. It's the application's responsibility to handle this when processing a block! | 2            |

Reviewer 3: ```suggestion
| block_max_gas   | int64 | Maximum sum of `GasWanted` in a proposed block. NOTE: blocks that violate this may be committed if there are Byzantine proposers. It's the application's responsibility to handle this when processing a block! | 2            |
```

Reviewer 3: I would consider taking the "NOTE" out of the table, for better readability

> 

Reviewer: Suggestion: "This parameters are used by the Proposed-Based TimeStamp (PBTS) algorithm", link to https://github.com/tendermint/tendermint/tree/master/spec/consensus/proposer-based-timestamp

| Name          | Type   | Description                   | Field Number |
|---------------|--------|-------------------------------|--------------|

> | precision     | [google.protobuf.Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration) | Bound for how skewed a proposer's clock may be from any validator on the network while still producing valid proposals. | 1            |

Reviewer: Skew is between clocks, correct? Between the proposer's clock and the clocks of any validator.

Author: Yes that is correct. Maybe it might be better to use the language "This sets a bound on how skewed a proposer's clock may be from any validator on the network while still producing valid proposals.". Do you think i should link to [proposer-based-timestamps](https://github.com/tendermint/tendermint/tree/331860c2a8c406fa89575165322b963925fb38c5/spec/consensus/proposer-based-timestamp) to be more clear?

Reviewer: Yes, this is a good idea. I would just remove "This sets a" from the sentence.

> | precision     | [google.protobuf.Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration) | This sets a bound on how skewed a proposer's clock may be from any validator on the network while still producing valid proposals. | 2            |

Reviewer: ```suggestion
| precision     | [google.protobuf.Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration) | Upper bound on how skewed a proposer's clock may be from any validator on the network while still producing valid proposals. | 2            |
```

> | message_delay | [google.protobuf.Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration) | Bound for how long a proposal message may take to reach all validators on a newtork and still be considered valid. | 2            |

Author: I swapped them according to what was reflective to the structure in the code see https://github.com/tendermint/tendermint/blob/27c523dccbe7f2f577868ee871b0042c1fef9a53/types/params.go#L85 @cmwaters

> | message_delay | [google.protobuf.Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration) | Bound for how long a proposal message may take to reach all validators on a newtork and still be considered valid. | 2            |

Reviewer 2: Right, your reference should actually be the proto file:

https://github.com/tendermint/tendermint/blob/27c523dccbe7f2f577868ee871b0042c1fef9a53/proto/tendermint/types/params.proto#L78

That's the holy grail

> | message_delay | [google.protobuf.Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration) | Bound for how long a proposal message may take to reach all validators on a newtork and still be considered valid. | 2            |

Reviewer 2: Protobuf is what we use for serialization but it's also, in a way, the language of the spec. If someone wanted to write Tendermint in Rust the specification requires them to adhere to the protobuf definitions (so that two implementations could talk to each other for example).

> ### TimeoutParams

Reviewer: The description for the timeouts was not introduced by this PR, but I have to say that I don't like it very much.

The parameters without `_delta` suffix are the initial values for the timeouts, increased by the parameter with `_delta` suffix at every new round of consensus.

Author: @cmwaters What are your thoughts on this one

| commit | [google.protobuf.Duration](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Duration) | Parameter that configures how long Tendermint will wait after receiving a quorum of precommits before beginning consensus for the next height.| 5 |
| bypass_commit_timeout | bool | Parameter that, if enabled, configures the node to proceed immediately to the next height once the node has received all precommits for a block, forgoing the commit timeout. |  6  |

> ## ABCIParams

Reviewer 2: ```suggestion
### ABCIParams
```

| Name                            | Type  | Description                                                                                      | Field Number |
|---------------------------------|-------|--------------------------------------------------------------------------------------------------|---|

> | vote_extensions_enable_height   | int64 |                                                                                                  | 1 |

Reviewer 4: [Here](https://github.com/tendermint/tendermint/blob/331860c2a8c406fa89575165322b963925fb38c5/spec/abci%2B%2B/abci%2B%2B_app_requirements.md#abciparamsvoteextensionsenableheight) is a description of the flag.
